### PR TITLE
Focused Launch: Load focused launch in Calypso for faster dev work.

### LINKED ISSUE #46842: Focused Launch: Temporarily move development to calypso
This will speed up development, as it won't require WBE/FSE running and updating in the background for every change

### PR DESCRIPTION
**Pull request by ciampo**
## Changes proposed in this Pull Request

When opening the Calypso block editor with the `gutenboarding/focused-launch-flow-calypso` flag enabled via the URL, render focused launch modal immediately.

This flag will also prevent the iframe from loading to prevent unnecessary overhead.

You can now dev happily without messing around with FSE/WBE/Sandboxing.

## Testing instructions

* Run `yarn start`.

**With flag enabled:**
* Go to `http://calypso.localhost:3000/post/YOUR_SITE_HERE.wordpress.com/home?flags=gutenboarding/focused-launch-flow-calypso`.
* It should load the focused launch modal.

**With flag disabled:**
* Go to `http://calypso.localhost:3000/post/YOUR_SITE_HERE.wordpress.com/home`.
* It should load the block editor as usual.

**Making sure FocusedLaunch doesn't break on FSE**
* Run `yarn dev --sync` on `apps/editing-toolkit`.
* Sandbox an unlaunched site created using start.
* Go to `http://calypso.localhost:3000/post/YOUR_UNLAUNCHED_SITE_CREATED_USING_START_HERE.wordpress.com/home?flags=gutenboarding/focused-launch-flow,gutenboarding/persistent-launch-button`.
* Focused launch should open up properly.

Closes #46842

### CODE REVIEW

File: client/landing/gutenboarding/onboarding-block/edit.tsx [hunk 1]
```diff
@@ -128,6 +129,14 @@ const OnboardingEdit: React.FunctionComponent< BlockEditProps< Attributes > > =
 				</Route>
 
 				<Route path={ makePath( Step.CreateSite ) }>{ createSiteOrError() }</Route>
+
+				<Route path={ makePath( Step.FocusedLaunchFlowModal ) }>
+					<FocusedLaunchModal
```
Comment (razvanpapadopol) [inline client/landing/gutenboarding/onboarding-block/edit.tsx:134]: Can we async load this only when some feature flag is enabled? At the moment it's adding 21Kb gzipped to `/new` entry-point.
Also, when the same feature flag is disabled we should probably return `<Redirect to={ makePath(Step.AcquireIntent) } />`

Comment (ciampo) [reply client/landing/gutenboarding/onboarding-block/edit.tsx:134]: All good points, I'll make sure to add them to this PR

**Review by wp-desktop — CHANGES_REQUESTED** [review]
WordPress Desktop CI Failure for job "wp-desktop-mac".

@ciampo please inspect this job's build steps for breaking changes at [this link](https://circleci.com/gh/Automattic/wp-calypso/935762). For temporal failures, you may try to "Rerun Workflow from Failed".

Please also ensure this branch is rebased off latest Calypso.

File: client/landing/gutenboarding/onboarding-block/edit.tsx (hunk 1 above)
Comment (yansern) [reply client/landing/gutenboarding/onboarding-block/edit.tsx]: - It no longer loads through gutenboarding's route, now load through calypso block editor route.
- It is now async loaded.

File: packages/launch/src/hooks/use-domain-suggestion.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ export function useDomainSuggestion() {
 			if ( ! domainSearch || domainSearch.length < 2 ) {
 				return;
 			}
-			return select( DOMAINS_STORE ).getDomainSuggestions( domainSearch, {
+			return select( DOMAIN_SUGGESTIONS_STORE ).getDomainSuggestions( domainSearch, {
```
Comment (yansern) [inline packages/launch/src/hooks/use-domain-suggestion.ts:20]: I see the `DOMAIN_SUGGESTIONS_STORE` being used more in other places even though `DOMAINS_STORE` is shorter.

File: client/landing/gutenboarding/onboarding-block/edit.tsx [hunk 3]
```diff
@@ -5,7 +5,7 @@ import * as React from 'react';
 import { Redirect, Switch, Route, useLocation } from 'react-router-dom';
 import type { BlockEditProps } from '@wordpress/blocks';
 import { useSelect } from '@wordpress/data';
-import { isEnabled } from 'config';
+import { isEnabled } from 'calypso/config';
```
Comment (yansern) [inline client/landing/gutenboarding/onboarding-block/edit.tsx:8]: This now becomes irrelevant to this PR, but let's just slip this in anyway.

File: packages/launch/src/stores.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+/**
+ * External dependencies
+ */
+import { Site, DomainSuggestions, Plans, Launch } from '@automattic/data-stores';
+
+// TODO: Check Site store registration discrepancies between:
```
Comment (yansern) [inline packages/launch/src/stores.tsx:6]: Adding a comment so I can open an issue from this.

File: packages/launch/src/stores.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+// - client/landing/gutenboarding/stores/site/index.ts
+// - apps/editing-toolkit/editing-toolkit-plugin/common/data-stores/site.ts
+const SITE_STORE = Site.register( { client_id: '', client_secret: '' } );
+
+const PLANS_STORE = Plans.register();
+
+// TODO: Check domain suggestions store registration discrepancies between:
+// - client/landing/gutenboarding/stores/domain-suggestions/index.ts
+// - apps/editing-toolkit/editing-toolkit-plugin/common/data-stores/domain-suggestions.ts
+const DOMAIN_SUGGESTIONS_STORE = DomainSuggestions.register( { vendor: 'variation2_front' } );
```
Comment (yansern) [inline packages/launch/src/stores.tsx:16]: Adding a comment so I can open an issue from this.

**Comment by ciampo** [comment]
Works correctly as per instructions. Code changes also LGTM, but since I'm a partial author of this PR, better wait for someone else to review.

**Review by yansern — APPROVED** [review]

**Pull request merged**